From 79a53d1633f44c2fd45ba8aa9e22cf20531ed659 Mon Sep 17 00:00:00 2001
From: tinywrkb <tinywrkb@gmail.com>
Date: Tue, 10 May 2022 21:19:24 +0300
Subject: [PATCH] cpan: Fix ba500568 regression, escape $ character

Regression was introduced by ba5005689bac0f5bcfc8a9531da3c576260ceef2
Without escaping $ character I'm getting the following error:

> Global symbol "$FLATPAK_DEST" requires explicit package name (did you forget to declare "my $FLATPAK_DEST"?) at /app/bin/flatpak-cpan-generator.pl line 125.
> Global symbol "$FLATPAK_DEST" requires explicit package name (did you forget to declare "my $FLATPAK_DEST"?) at /app/bin/flatpak-cpan-generator.pl line 125.
> Bareword "main" not allowed while "strict subs" in use at /app/bin/flatpak-cpan-generator.pl line 140.
> Execution of /app/bin/flatpak-cpan-generator.pl aborted due to compilation errors.
---
 cpan/flatpak-cpan-generator.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cpan/flatpak-cpan-generator.pl b/cpan/flatpak-cpan-generator.pl
index 08436f1..cd0e837 100755
--- a/cpan/flatpak-cpan-generator.pl
+++ b/cpan/flatpak-cpan-generator.pl
@@ -122,7 +122,7 @@ sub main {
       "    mod_dir=\$1",
       "    cd \$mod_dir",
       "    if [ -f 'Makefile.PL' ]; then",
-      "        perl Makefile.PL PREFIX=${FLATPAK_DEST} && make install PREFIX=${FLATPAK_DEST}",
+      "        perl Makefile.PL PREFIX=\${FLATPAK_DEST} && make install PREFIX=\${FLATPAK_DEST}",
       "    elif [ -f 'Build.PL' ]; then",
       "        perl Build.PL && ./Build && ./Build install",
       "    else",
